src/sdk/pynni/nni/smartparam.py

```diff
@@ -115,7 +119,51 @@ def qlognormal(mu, sigma, q, name=None, key=None):
     def function_choice(funcs, name=None, key=None):
         return funcs[_get_param(key)]()
 
+    def mutable_layer(
+            mutable_id,
+            mutable_layer_id,
+            funcs,
+            funcs_args,
+            fixed_inputs,
+            optional_inputs,
+            optional_input_size=0,
+            debug_mode=False,
+            name=None,
+            key=None):
```

Comment:
It seems 'debug_mode', 'name' and 'key' are not used?  can they be removed?

Reply:
Quanlu wants to add this features in the future so the API is designed in advanced. But anyway I will remove this now cuz there's no plan for these features.

Reply:
Sure, the parameters can be added in the future when it is needed, no need to add them in advance.